mesonbuild/compilers/cpp.py

```diff
@@ -268,17 +268,13 @@ def get_options(self) -> 'MutableKeyedOptionDictType':
     def get_option_compile_args(self, target: 'BuildTarget', env: 'Environment', subproject: T.Optional[str] = None) -> T.List[str]:
         args: T.List[str] = []
 
-        std = self.get_compileropt_value('std', env, target, subproject)
         rtti = self.get_compileropt_value('rtti', env, target, subproject)
         debugstl = self.get_compileropt_value('debugstl', env, target, subproject)
         eh = self.get_compileropt_value('eh', env, target, subproject)
 
-        assert isinstance(std, str)
         assert isinstance(rtti, bool)
         assert isinstance(eh, str)
         assert isinstance(debugstl, bool)
-        if std != 'none':
-            args.append(self._find_best_cpp_std(std))
```

Comment:
In general, it looks like this will mean moving -std=c++XXXX from before various `-D` / eh / rtti options, to after it, which should be fine as order shouldn't matter here.